inbox/mailsync/backends/imap/generic.py

```diff
@@ -534,35 +534,37 @@ def poll_for_changes(self):
             log.debug("polling for changes")
             self.poll_impl()
 
-    def create_message(self, db_session, acct, folder, msg):
-        assert acct is not None and acct.namespace is not None
+    def create_message(
+        self, db_session: Any, account: Account, folder: Folder, raw_message: RawMessage
+    ) -> Optional[ImapUid]:
+        assert account is not None and account.namespace is not None
 
         # Check if we somehow already saved the imapuid (shouldn't happen, but
         # possible due to race condition). If so, don't commit changes.
         existing_imapuid = (
             db_session.query(ImapUid)
             .filter(
-                ImapUid.account_id == acct.id,
+                ImapUid.account_id == account.id,
                 ImapUid.folder_id == folder.id,
-                ImapUid.msg_uid == msg.uid,
+                ImapUid.msg_uid == raw_message.uid,
             )
             .first()
         )
         if existing_imapuid is not None:
-            log.error(
+            log.warning(
                 "Expected to create imapuid, but existing row found",
-                remote_msg_uid=msg.uid,
+                remote_msg_uid=raw_message.uid,
                 existing_imapuid=existing_imapuid.id,
             )
             return None
 
         # Check if the message is valid.
-        if msg.body is None:
+        if not raw_message.body:
```

Comment:
I guess those two will be always equivalent? Can a message be valid when `raw_message.body` is `""`?

Reply:
it can be actually `""` but never `None`. As soon as put typing info mypy catched this because this code was unreachable and we should never process empty messages.